examples/synthea-breast-cancer/Ali_Laiba_syntheaEDA/SyntheaClassifiers.ipynb

```diff
@@ -0,0 +1,1710 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='KNNAnalysis'></a>\n",
+    "**Analysing our Results**\n",
+    "\n",
+    "We notice that the KNN classifier, with k = 7, is quite an accurate model, with a balanced accuracy of 0.9154 and a macro f1 score of 0.9117. We also have an MCC value of 0.9088, indicating that our model is quite accurate in its predictions, despite the imbalanced class sizes. Thus, as a whole, the model is quite precise and error-free, and also doesn't produce too many false positives and false negatives. However, we will come to see later on that this model is not as good as many of the models we will discuss later on. \n",
+    "\n",
+    "On the other hand, the KNN classifier is quite easy to picture, in that we intuitively understand that it looks to find the k-nearest points and categorizes our sample based on the given data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='DTClassifier'></a>\n",
+    "**Building a Decision Tree Classifier**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tree_classifier = DecisionTreeClassifier(criterion='gini', max_depth=5).fit(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='DTResults'></a>\n",
+    "**Viewing Decision Tree Results**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CONFUSION MATRIX:\n",
+      "[[137   3   0   0]\n",
+      " [  1  99   0   0]\n",
+      " [  0   0  31   0]\n",
+      " [  0   0   0   9]]\n",
+      "\n",
+      "CLASSIFICATION REPORT:\n",
+      "              precision    recall  f1-score   support\n",
+      "\n",
+      "     Stage 1       0.99      0.98      0.99       140\n",
+      "     Stage 2       0.97      0.99      0.98       100\n",
+      "     Stage 3       1.00      1.00      1.00        31\n",
+      "     Stage 4       1.00      1.00      1.00         9\n",
+      "\n",
+      "    accuracy                           0.99       280\n",
+      "   macro avg       0.99      0.99      0.99       280\n",
+      "weighted avg       0.99      0.99      0.99       280\n",
+      "\n",
+      "\n",
+      " Balanced Accuracy  Macro F1 Score      MCC\n",
+      "          0.992143        0.991452 0.976671\n",
+      "\n",
+      "AVERAGE CROSS-VALIDATION SCORE: 0.9657142857142856\n"
+     ]
+    }
+   ],
+   "source": [
+    "dt_pred = tree_classifier.predict(X_test)\n",
+    "print_results(y_test, dt_pred)\n",
+    "print_cross_validation(DecisionTreeClassifier(criterion='gini', max_depth=5), X, y)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Visualizing Tree**\n",
```

Comment:
This section is really a great use of the Jupyter Notebook format, getting to walk through the code used to generate the model and visualize its internals is super cool.